src/figuresClasses.ts

```diff
@@ -1,11 +1,97 @@
-export interface Figure {}
+type Shape = `triangle` | `circle` | `rectangle`;
+type Color = `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  public color: Color;
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
[CODE KNOWLEDGE] - You don't need to manually assign parameters to properties if you use TypeScript's parameter properties. You can declare the properties directly in the constructor signature with access modifiers to achieve the same effect. For example: `constructor(public color: Color, private a: number, private b: number, private c: number) { ... }`

Reply:
> [CODE KNOWLEDGE] - You don't need to manually assign parameters to properties if you use TypeScript's parameter properties. You can declare the properties directly in the constructor signature with access modifiers to achieve the same effect. For example: `constructor(public color: Color, private a: number, private b: number, private c: number) { ... }`

Good point, update in all places